docs/requirements.md

```diff
@@ -0,0 +1,49 @@
+At its heart ServiceControl and ServicePulse provide operational support for systems built on the Particular Service Platform. They do this by:
+
+- monitoring system health
+- diagnosing and treating errors
+
+In this context, system refers to NServiceBus endpoints and communication with third-party services.
+
+ServiceControl should provide the following features:
+
+- process and store the contents of the error queue
+- replay messages from the error queue
+- process and store the contents of the audit queue
+- generate and collect heartbeat messages
+- generate and collect saga state changes
+- provide an API for monitoring user-defined conditions
+- provide operational events for consumption by other applications
+
+Implicit in these requirements is that ServiceControl should:
+
+- scale to an entire system based on the Particular Service Platform
+- be durable (i.e. no message loss)
+
+### Process and store the contents of the error queue
+
+Rationale: To view the contents of the error queue in order to triage potential bugs or infrastructure failures in a system.
+
+### Replay messages from the error queue
+
+Rationale: To recover from bugs or infrastructure failures in order to maintain stability of a system.
+
+### Process and store the contents of the audit queue
+
+Rationale: To verify that a replayed message completed successfully. To understand the emergent design of a system through correllated messages. To identify design flaws in a system.
+
+### Generate and collect heartbeat messages
+
+Rationale: Verify that endpoints are up and able to send messages. Facilitates notification in the event an endpoint is no longer operational.
+
+### Generate and collect saga state changes
+
+Rationale: To understand the operational behavior of a long-running process and to troubleshoot design flaws.
+
+### Provide an API for monitoring user-defined conditions
```

Comment:
Well, in theory there is a difference between "the Sales DB is down" vs "the Sales endpoint cannot connect to the Sales DB" and Custom Checks lets you develop the latter trivially. If we ever put the control channel in I would love to see endpoints expose Custom Checks to SC but SC control when/how they are run so we could have an interface in SP to manage it. i.e. Click an endpoint, get a list of checks to run, click the "Check Connection to Sales DB" button, get back a response saying "Success!". Then the whole concept of periodic checks goes away and you put the scheduling part in SC as it's an operational automation concern. Developers just expose whatever checks they feel are appropriate.